contracts/components/utils/Routed.sol

```diff
@@ -4,43 +4,22 @@
 pragma solidity ^0.8.9;
 
 import "./AccessManaged.sol";
-import "../router/IRouter.sol";
-import "../../errors/GeneralErrors.sol";
 
+/**
+ * Since Router is deprecated, we are keeping RoutedUpgradeable in this state to preserve storage
+ * layout in deployed `BaseComponentUpgradeable` contracts.
+ */
 abstract contract RoutedUpgradeable is AccessManagedUpgradeable {
-    IRouter private _router;
 
-    event RouterUpdated(address indexed router);
-
-    /**
-     * @notice Initializer method, access point to initialize inheritance tree.
-     * @param router address of Routed.
-     */
-    function __Routed_init(address router) internal initializer {
-        if (router == address(0)) revert ZeroAddress("router");
-        _router = IRouter(router);
-        emit RouterUpdated(router);
-    }
+    /// @custom:oz-renamed-from router
+    address private _deprecated_router;
 
-    /**
-     * @dev Routed contracts can use this method to notify the subscribed observers registered
-     * in Router routing table. Hooks may revert on failure or not, as set when calling
-     * Router.setRoutingTable(bytes4 sig, address target, bool enable, bool revertsOnFail)
-     * @param data keccak256 of the method signature and param values the listener contracts.
-     * will execute.
-     */
-    function _emitHook(bytes memory data) internal {
-        if (address(_router) != address(0)) {
-            try _router.hookHandler(data) {}
-            catch {}
-        }
-    }
+    event RouterUpdated(address indexed router);
 
-    /// Sets new Router instance. Restricted to DEFAULT_ADMIN_ROLE.
-    function setRouter(address newRouter) public onlyRole(DEFAULT_ADMIN_ROLE) {
-        if (newRouter == address(0)) revert ZeroAddress("newRouter");
-        _router = IRouter(newRouter);
-        emit RouterUpdated(newRouter);
+    /// Sets Router instance to address(0). Restricted to DEFAULT_ADMIN_ROLE.
+    function disableRouter() public onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
Why is this necessary? If for cleanup, can we do it in a reinitializer? (Related: #127)

Reply:
It is for cleanup.
We can and we should use reinitializers (now that you kindly reminded me that they are a thing), but in this version of OZ upgradeable, `_version` is not exposed and all of the components inherit from `Router`. 

I would not know how to call the `reinitializer` with a number that works for all of them, so I would let `disableRouter` stand in this PR, and continue the discussion in #127 . I'm open to upgrading OZ to 4.8 (if it's safe) before the code is audited to implement this properly.

Does that look good to you?

Reply:
>I would not know how to call the reinitializer with a number that works for all of them

This is a good point. Ok with leaving it like this but would slightly prefer removing `onlyRole` and reverting if the router is already zero.

Reply:
5fed729